src/main/java/appeng/util/inv/WrapperInvItemHandler.java

```diff
@@ -0,0 +1,156 @@
+
+package appeng.util.inv;
+
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+
+public class WrapperInvItemHandler implements IInventory
+{
+	final private IItemHandlerModifiable inv;
+
+	public WrapperInvItemHandler( final IItemHandlerModifiable inv )
+	{
+		this.inv = inv;
+	}
+
+	@Override
+	public String getName()
+	{
+		return null;
+	}
+
+	@Override
+	public boolean hasCustomName()
+	{
+		return false;
+	}
+
+	@Override
+	public ITextComponent getDisplayName()
+	{
+		return null;
+	}
+
+	@Override
+	public int getSizeInventory()
+	{
+		return inv.getSlots();
+	}
+
+	@Override
+	public boolean isEmpty()
+	{
+		for( int x = 0; x < inv.getSlots(); x++ )
+		{
+			if( !inv.getStackInSlot( x ).isEmpty() )
+			{
+				return false;
+			}
+		}
+		return true;
+	}
+
+	@Override
+	public ItemStack getStackInSlot( int index )
+	{
+		return inv.getStackInSlot( index );
+	}
+
+	@Override
+	public ItemStack decrStackSize( int index, int count )
+	{
+		return inv.extractItem( index, count, false );
+	}
+
+	@Override
+	public ItemStack removeStackFromSlot( int index )
+	{
+		return inv.extractItem( index, inv.getSlotLimit( index ), false );
+	}
+
+	@Override
+	public void setInventorySlotContents( int index, ItemStack stack )
+	{
+		inv.setStackInSlot( index, stack );
+	}
+
+	@Override
+	public int getInventoryStackLimit()
+	{
+		int max = 0;
+		for( int i = 0; i < inv.getSlots(); ++i )
+		{
+			max = Math.max( max, inv.getSlotLimit( i ) );
+		}
+		return max;
+	}
+
+	@Override
+	public void markDirty()
+	{
+	}
+
+	@Override
+	public boolean isUsableByPlayer( EntityPlayer player )
+	{
+		return false;
+	}
+
+	@Override
+	public void openInventory( EntityPlayer player )
+	{
+	}
+
+	@Override
+	public void closeInventory( EntityPlayer player )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1186)